solr/core/src/java/org/apache/solr/cli/HealthcheckTool.java

```diff
@@ -170,7 +172,15 @@ protected void runCloudTool(CloudSolrClient cloudSolrClient, CommandLine cli) th
           q = new SolrQuery("*:*");
           q.setRows(0);
           q.set(DISTRIB, "false");
-          try (var solrClientForCollection = SolrCLI.getSolrClient(coreUrl)) {
+
+          // SolrCLI.getSolrClient converts the coreUrl back into a root Solr url, but we need to
+          // talk to specific core so manually creating the client.
```

Comment:
Hmm, would you say this is a bug, or is it well documented in the javadoc?

Reply:
I looked a bit more at the code, and decided to go with the javadoc approach!